internal/client.go

```diff
@@ -713,6 +725,22 @@ type (
 		callbacks []*commonpb.Callback
 	}
 
+	// StartWorkflowOperation is a type of operation that can be executed as part of a workflow start.
+	StartWorkflowOperation interface {
+		isStartWorkflowOperation()
+	}
+
+	// UpdateWorkflowOperation is used to perform Update-with-Start.
+	// See PrepareUpdateWorkflowOperation for details.
+	UpdateWorkflowOperation struct {
+		input     *ClientUpdateWorkflowInput
+		executing atomic.Bool
+		done      atomic.Bool
+		doneCh    chan struct{}
+		handle    WorkflowUpdateHandle
+		err       error
```

Comment:
May want a comment making it clear these fields should not be accessed until after `doneCh` is received from